### DISCUSSION TITLE
Custom logic for the artisan command which triggers the job

### DISCUSSION THREAD

**Discussion by tpetry** [spatie/laravel-artisan-dispatchable#3] [Ideas]
Currently arguments to a job have to be provided directly for the constuctor parameter names:
```cli
php artisan my-job --my-first-argument="My string value"
```

It would be really nice if the there could be a special method which would behave like artisan's `handle` method which has to return the job but is able to do a lot more:
```php
class MyJob implements ArtisanDispatchable
{
    public function __construct(
        string $myFirstArgument, 
    ) {}

    public function handle()
    {
        // perform some work...
    }

    public function buildJob(Command $command)
    {
        // $command instanceof \Illuminate\Console\Concerns\InteractsWithIO
        $command->info('Starting job');
        return new self($command->argument('myarg'));
    }
}


```

**Comment by freekmurze** [comment]
Nice idea. I think if we add this, then a function like `buildJob` should be static, as we can new up an instance that requires arguments to be passed to it's constructor.

**Comment by tpetry** [comment]
Oh, the more i think about it the return value could even be an array of jobs.
